src/NServiceBus.RavenDB/Internal/SessionExtensions.cs

```diff
@@ -0,0 +1,16 @@
+namespace NServiceBus.Persistence.RavenDB
+{
+    using Raven.Client.Documents.Session;
+
+    static class SessionExtensions
+    {
+        public static void StoreVersionInMetadata(this IAsyncDocumentSession session, object entity)
+        {
+            var metadata = session.Advanced.GetMetadataFor(entity);
+            // TODO discuss what to makes most sense
+            metadata[NServiceBusPersistenceRavenDBVersionMetadataKey] = GitVersionInformation.AssemblySemVer;
+        }
+
+        internal const string NServiceBusPersistenceRavenDBVersionMetadataKey = "NServiceBus-Persistence-RavenDB-Version";
```

Comment:
RavenDB uses the dash convention for their stuff so I sticked to that as well, i.ex.

- `@change-vector`
- `Raven-Clr-Type`